### PR TITLE
feat: add rejection to contract approval workflow

### PR DESCRIPTION
**Pull request by volkflo**
Closes: https://github.com/apiman/apiman/issues/1679

### CODE REVIEW

File: manager/ui/war/plugins/api-manager/ts/api/api-contracts.ts [hunk 1]
```diff
@@ -30,16 +30,66 @@ _module.controller("Apiman.ApiContractsController",
             })
         });
 
-        $scope.approveContract = (contract: ContractSummaryBean) => {
+        let handleApprovalError = function (err) {
+            if (err !== 'cancel') {
+                console.error(err)
```
Comment (msavy) [inline manager/ui/war/plugins/api-manager/ts/api/api-contracts.ts:35]: Please consider using `Logger` for this. Looks like it's already injected, so should be usable directly.

Logger.debug or something

**Review by msavy — DISMISSED** [review]
Some small nits to resolve, but overall looks good.

Thanks for collaborating on this!

File: distro/conf/src/main/resources/notifications/email/tpl/de/apiman.client.contract.approval.rejected.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<!--
```
[the remark this answers is not in the document]
Comment (volkflo) [reply distro/conf/src/main/resources/notifications/email/tpl/de/apiman.client.contract.approval.rejected.html:1]: Funny, Intellij added that but only for that single file not for the others :D

File: distro/conf/src/main/resources/notifications/email/notification-template-index.json [hunk 3]
```diff
@@ -17,6 +17,20 @@
       "html": "tpl/en/apiman.client.contract.approval.granted.html",
       "plain": "tpl/en/apiman.client.contract.approval.granted.txt",
       "category": "API_ADMINISTRATION"
+    },
+    "apiman.client.contract.approval.rejected": {
+      "subject": "Your request to sign up to API {event.apiId}/{event.apiVersion} has been rejected!",
```
[the remark this answers is not in the document]
Comment (volkflo) [reply distro/conf/src/main/resources/notifications/email/notification-template-index.json:22]: Same in German, will change it :)

File: distro/conf/src/main/resources/notifications/email/notification-template-index.json [hunk 4]
```diff
@@ -17,6 +17,20 @@
...
+      "html": "tpl/en/apiman.client.contract.approval.rejected.html",
+      "plain": "tpl/en/apiman.client.contract.approval.rejected.txt",
+      "category": "API_ADMINISTRATION"
+    }
+  },
+  "de": {
```
[the remark this answers is not in the document]
Comment (msavy) [reply distro/conf/src/main/resources/notifications/email/notification-template-index.json:28]: Okay, then let's leave it for now!

File: manager/ui/war/plugins/api-manager/html/directives/getvalueModal.html [hunk 5]
```diff
@@ -1,25 +0,0 @@
-      <div class="modal fade" id="valueModal" tabindex="-1" role="dialog" aria-hidden="true">
```
[the remark this answers is not in the document]
Comment (volkflo) [reply manager/ui/war/plugins/api-manager/html/directives/getvalueModal.html:1]: At least I think so, lets hope that I did not accidentally broke something with that again :laughing:

File: distro/conf/src/main/resources/notifications/email/notification-template-index.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (msavy) [reply distro/conf/src/main/resources/notifications/email/notification-template-index.json]: Great, thanks!

**Review by msavy — APPROVED** [review]

**Pull request merged**